selfdrive/thermald/thermald.py

```diff
@@ -50,10 +50,25 @@
 
 prev_offroad_states: Dict[str, Tuple[bool, Optional[str]]] = {}
 
+tz_by_type: Optional[Dict[str, int]] = None
+def populate_tz_by_type():
+  global tz_by_type
+  tz_by_type = {}
+  for n in os.listdir("/sys/devices/virtual/thermal"):
+    if not n.startswith("thermal_zone"):
+      continue
+    with open(os.path.join("/sys/devices/virtual/thermal", n, "type")) as f:
+      tz_by_type[f.read().strip()] = int(n.lstrip("thermal_zone"))
+
 def read_tz(x):
   if x is None:
     return 0
 
+  if isinstance(x, str):
```

Comment:
think it's cleaner to do this in `HARDWARE.get_thermal_config`, then it always returns a path you can read from

Reply:
I disagree with the way it's currently structured now. The numbers are meaningless and should imo be handled at the last possible moment.

Reply:
That's probably better

Reply:
Ok, to merge like this. I can do the refactor when I'm back from holiday.